### PR TITLE
[Enhancement] Add more legendary battle music; add music preference setting

### PR DESCRIPTION
**Pull request by Madmadness65**
## What are the changes?

This PR adds new Legendary battle music for a vast amount of the Legendary and Mythical Pokémon, as well as a new setting to toggle all of this new music. The new setting, labelled "Music Preference", has a toggle between "Consistent" (Prefer Gen 5 music, identical to current version, selected by default) and "Mixed" (Disregard generation, prefer variety), and is found in the Audio Settings.

## Why am I doing these changes?

More music, specifically battle music, has been desired for quite a long time. Although the initial concerns were that it clashes with the existing Gen 5 and PMD music, offering it as a user choice seemed like the most reasonable solution.

## What did change?

A large assortment of new music was added. This is all the music in order:
**HGSS**
- Raikou Battle
- Entei Battle
- Suicune Battle
- Lugia Battle
- Ho-oh Battle
- Arceus Battle

**XY**
- Xerneas & Yveltal Battle
- Kanto Legendary Battle

**ORAS**
- Legendary Titan Battle (Replaces the B2W2 equivalent track when the Mixed option is selected)
- Groudon & Kyogre Battle
- Rayquaza Battle
- Deoxys Battle
- Lake Guardians Battle
- Sinnoh Legendary Battle
- Dialga & Palkia Battle
- Giratina Battle

**SM**
- Tapu Battle
- Solgaleo & Lunala Battle
- Ultra Beast Battle

**USUM**
- Dusk Mane & Dawn Wings Necrozma Battle (Currently unused)
- Ultra Necrozma Battle (Currently unused, planned use in the future)

**SWSH**
- Zacian & Zamazenta Battle
- Glastrier & Spectrier Battle
- Calyrex Battle
- Galarian Legendary Birds Battle

**SV**
- Treasures of Ruin Battle
- Loyal Three Battle
- Ogerpon Battle
- Terapagos Battle
- Pecharunt Battle

Additionally, one new entry to the Audio Settings was added, and it's use is primarily within the `battle.ts` file.
### Screenshots/Videos

https://github.com/pagefaultgames/pokerogue/assets/59298170/10ee3169-89f3-441f-8904-741a94412d7a

## How to test the changes?

Edit the opponent override to any Legendary or Mythical Pokémon associated with these tracks, and start/resume a run. The option can be toggled freely within the game's audio settings and will reload the game so the switched music can play.

## Checklist
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Comment by sirzento** [comment]
Yeah ok that is awesome to hear the original music from those fights

File: src/battle.ts [hunk 1]
```diff
@@ -209,22 +209,122 @@ export default class Battle {
...
+        if (scene.musicPreference === 0) {
+          if (pokemon.species.speciesId === Species.REGIROCK || pokemon.species.speciesId === Species.REGICE || pokemon.species.speciesId === Species.REGISTEEL || pokemon.species.speciesId === Species.REGIGIGAS || pokemon.species.speciesId === Species.REGIELEKI || pokemon.species.speciesId === Species.REGIDRAGO) {
+            return "battle_legendary_regis_g5";
+          }
+          if (pokemon.species.speciesId === Species.COBALION || pokemon.species.speciesId === Species.TERRAKION || pokemon.species.speciesId === Species.VIRIZION || pokemon.species.speciesId === Species.TORNADUS || pokemon.species.speciesId === Species.THUNDURUS || pokemon.species.speciesId === Species.LANDORUS || pokemon.species.speciesId === Species.KELDEO || pokemon.species.speciesId === Species.MELOETTA || pokemon.species.speciesId === Species.GENESECT) {
+            return "battle_legendary_unova";
+          }
+          if (pokemon.species.speciesId === Species.RESHIRAM || pokemon.species.speciesId === Species.ZEKROM) {
+            return "battle_legendary_res_zek";
+          }
```
Comment (Tempo-anon) [inline src/battle.ts:221]: Is this check needed when reshiram and zekrom will hit line 225 for the same effect?

**Review by Tempo-anon — DISMISSED** [review]
Functionally it works but I wonder if a `switch` would make be a bit more readable

File: src/battle.ts (hunk 1 above)
Comment (Madmadness65) [reply src/battle.ts:221]: This is true, I can remove 219-221 and 275-277 and the same effect will be achieved.

**Comment by Madmadness65** [comment]
> Functionally it works but I wonder if a `switch` would make be a bit more readable

It was just a result of copying the code that already existed in `battle.ts`. If a better way of cleaning it up can be done at some point, that'd be great.

**Review by Tempo-anon — APPROVED** [review]

**Pull request merged**